### ISSUE TITLE
Cypress change visiting url with adding `/__/#/tests/__all`

### ISSUE THREAD

**Issue by OksanaFirtych** [cypress-io/cypress#4453]
Cypress adds some unexpected suffix in testing url.

### Current behavior:
1. `cy.visit('http://localhost:3000/')`
2. run cypress.
3. it openes url `http://localhost:3000/__/#/tests/__all`
4. Page not found

1. `cy.visit('http://localhost:3000/simple')`
2. run cypress.
3. it openes url `http://localhost:3000/__/#/tests/__all`
4. Page not found

### Desired behavior:

Should visit exactly url from script http://localhost:3000/ or http://localhost:3000/simple and test rendered page.

### Steps to reproduce: (app code and test code)

I've created testing repository. https://github.com/OksanaFirtych/test_cypress

1. Clone repository
2. npm run start
Visit http://localhost:3000/ or http://localhost:3000/simple in browes to verify that it's started
3. npm run cy:open
4. Run all specs or main.spec.js

### Versions

Cypress 3.2.0
Chrome 75.0.3770.80

**Comment by OksanaFirtych** [comment]
Could somebody confirm that it existing bug or say what I'm doing wrong, please? This quite critical thing doesn't allow me to move forward.

**Comment by rizsky** [comment]
same as i did. But it run smoothly when it headless or when its browser is electron. When its on chrome 😕 
:
![image](https://user-images.githubusercontent.com/43275779/59505779-9e331380-8ed0-11e9-9ef1-e17f9e37b3ee.png)

**Comment by flotwig** [comment]
@OksanaFirtych 	Cypress works by opening your tests in a frame inside of a URL that looks like `http://<your testing domain>/__/#/tests/<tests that are running>`, so this in itself is not an issue. Is your site not loading properly?

**Comment by OksanaFirtych** [comment]
@flotwig Yes, my site with routing isn't loading at all any pages. But if I remove routing at all it will work under cypress. As you can see above it's not only mine problem.

**Comment by flotwig** [comment]
@OksanaFirtych I see, can you share a screenshot of the failure and a copy of the entire `DEBUG` log? Here are some instructions: https://docs.cypress.io/guides/guides/debugging.html#Print-DEBUG-logs

**Comment by OksanaFirtych** [comment]
@flotwig it looks like nothing. 
<img width="1280" alt="Screenshot 2019-06-17 at 22 04 25" src="https://user-images.githubusercontent.com/31488378/59629648-017caa00-914c-11e9-9ed6-0e406c948c1f.png">

[cy.log](https://github.com/cypress-io/cypress/files/3298468/cy.log)

**Comment by flotwig** [comment]
@OksanaFirtych Can you share a screenshot of `chrome://policy` from the browser that Cypress launches? This looks similar to #1253.

Does it work if you run it in Electron?

**Comment by OksanaFirtych** [comment]
Yes, it works with Electron

![Screenshot 2019-06-18 at 09 28 10](https://user-images.githubusercontent.com/31488378/59658224-b2666180-91ab-11e9-95ee-5994cd11ae98.png)
![Screenshot 2019-06-18 at 09 28 43](https://user-images.githubusercontent.com/31488378/59658227-b4c8bb80-91ab-11e9-80d0-9a6972350a27.png)

**Comment by flotwig** [comment]
I see `ProxyMode: pac_script` in your settings, which unfortunately means you are being affected by #1253 - your administrator has settings that are preventing Cypress from launching Chrome. We can't do much about this from our end. You can check [this comment](https://github.com/cypress-io/cypress/issues/1253#issuecomment-465782037) for some known workarounds.

Closing as duplicate of #1253

**Comment by AbreezaSaleem** [comment]
I have a url like this as well but when I try to hit a url with a different domain it gives me something like 

```
https://twitter.com/__/#/tests/integration/folder/test_spec.js
```

And since it is an invalid url, my application reroutes to the default url (`localhost:3000/users)` and for some reason the whole `test_spec.js` runs again. And when it reaches the test that `visits` the `twitter` url it again reroutes to the default url and again the tests in the `test_spec.js` start running again and the cycle continues.

Also, I have defined two `describe` blocks in the spec file which means I wont be getting the `same origin` error. 

Can someone please tell me how I can _just_ visit the `https://twitter.com` link when using `cypress.visit`?

**Comment by jennifer-shehane** [comment]
A fix for this is available starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments) which you can access by setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

The fix is experimental, so there may be some situations where the this is not fixed. 

If you're still this issue while setting the `experimentalSourceRewriting` to true in 4.6.0 - [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a reproducible example + screenshots, etc - filling out our issue template.
